### PR TITLE
Roll test262-parser-tests

### PR DESCRIPTION
**Pull request by bakkot**
Including some refactoring of the test harness to make this easier.

### CODE REVIEW

File: test/test262/errors.js [hunk 1]
```diff
@@ -11,123 +12,109 @@ function parse(src, isModule, earlyErrors) {
...
+        let passTree, passLocations;
+        expect(() => {
+          ({ tree: passTree, locations: passLocations } = parse(passSrc, f.match('.module.js'), true));
+        }).to.not.throwError();
+        locationSanityCheck(passTree, passLocations);
+
+        let passExplicitSrc = fs.readFileSync(passExplicitTestFile, 'utf8');
+        let passExplicitTree;
+        expect(() => {
+          passExplicitTree = parse(passExplicitSrc, f.match('.module.js'), true).tree;
```
Comment (michaelficarra) [inline test/test262/errors.js]: Can you extract all of these `f.match` calls to an `isModule` function?

**Review by michaelficarra — APPROVED** [review]
LGTM otherwise.

File: test/test262/errors.js (hunk 1 above)
Comment (bakkot) [reply test/test262/errors.js]: Done.

**Review by michaelficarra — APPROVED** [review]

File: test/test262/errors.js [hunk 2]
```diff
@@ -2,132 +2,123 @@ import fs from 'fs';
...
 let scriptDir = 'node_modules/test262-parser-tests';
 
-function parse(src, isModule, earlyErrors) {
-  return (isModule ? parseModuleWithLocation : parseScriptWithLocation)(src, { earlyErrors });
+function parse(src, asModule, earlyErrors) {
+  return (asModule ? parseModuleWithLocation : parseScriptWithLocation)(src, { earlyErrors });
+}
+
+function isModule(f) {
+  return /.module.js/.test(f);
```
Comment (michaelficarra) [inline test/test262/errors.js]:
```js
return /\.module\.js/.test(f);
```

Comment (bakkot) [reply test/test262/errors.js]: 😆

**Pull request merged**